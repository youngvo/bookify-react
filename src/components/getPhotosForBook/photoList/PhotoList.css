.photo-list-content {
    /*top: 5px;
    left: 50px;
    right: 0px;
    bottom: 5px;
    position: absolute;*/
    overflow-y: auto;
    float:left;
    display:inline-block;
}

.content-left-content{
    top: 0px;
    left: 220px;
    right: 115px;
    bottom: 0;
    display: inline-block;
    position: absolute;
    overflow-y: auto;
    margin-left: -20px;
    margin-top: -15x;
}

.scrollbar-style::-webkit-scrollbar-track{
    /*-webkit-box-shadow: inset 1 1 1px rgba(0,0,0,1);*/
	width:12px;
    border-radius: 15px;
	background:rgba(122,122,122,0.3);
}

.scrollbar-style::-webkit-scrollbar-thumb{
    border-radius: 15px;
	/*-webkit-box-shadow: inset 0 0 6px rgba(0,0,0,.3);*/
	background: linear-gradient(to right, #d0d0d0, #e7e7e7);
    border: 1px solid white;
}

.scrollbar-style::-webkit-scrollbar{
	width: 15px;
	background:transparent;
}
