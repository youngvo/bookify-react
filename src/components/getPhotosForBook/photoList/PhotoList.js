import React from 'react';

import './PhotoList.css';
import PhotoCell from '../photoCell/PhotoCell';

const renderPhotoList = (photos, onPhotoCellClick, isSelected, iconClick) => (
    photos.map((photo, index) => (
      <PhotoCell
          isSelected={isSelected}
          key={photo.id}
          photo={photo}
          onClick={onPhotoCellClick}
          iconClick={iconClick}
      />
  ))
);

const PhotoList = ({ photos, posX, posY, scaleX, scaleY, onPhotoCellClick, isSelected, iconClick}) => (
    <div className="photo-list-content">
        {renderPhotoList(photos, onPhotoCellClick, isSelected, iconClick)}
    </div>
);

export default PhotoList;
