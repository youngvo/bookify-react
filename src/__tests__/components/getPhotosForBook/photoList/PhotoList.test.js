import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import PhotoList from './../../../../components/getPhotosForBook/photoList/PhotoList';
//import PhotoCell from './../../../../components/getPhotosForBook/photoCell/PhotoCell';
describe('test PhotoList component', () => {
    const photo = {
            id: '',
            name: '',
            picUrl: '',
            imageUrl: '',
            width: 0,
            height: 0,
            createdTime: '',
            updatedTime: '',
            albumId: ''
        };
    let photos = [{id:1}, {id:2}];
    let posX, posY, scaleX, scaleY ; 
    let onPhotoCellClick;
    let photoList;
    beforeEach(() => {
        onPhotoCellClick = jest.fn();
        photoList = shallow(<PhotoList photos={photos} onPhotoCellClick={onPhotoCellClick} />);
    });

    describe('PhotoList snapshot', () => {
        it('PhotoList does not change layout', () => {
            const component = renderer.create(<PhotoList photos={photos} onPhotoCellClick={onPhotoCellClick} />);
            const tree = component.toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('PhotoList DOM test', () => {

        it('PhotoList contains div with className="photo-list-content"', () => {
            expect(photoList.find('div.photo-list-content').length).toEqual(1);
        });

        it('renderPhotoList return right', () => {
            expect(photoList.find('PhotoCell').length).toEqual(2);
        })
    })
});