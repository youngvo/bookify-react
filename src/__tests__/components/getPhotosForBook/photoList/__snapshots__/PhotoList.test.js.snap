exports[`test PhotoList component PhotoList snapshot PhotoList does not change layout 1`] = `
<div
  className="photo-list-content">
  <div
    className="selecting-photo-cell"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    <div
      className="preview-photo-zone">
      <img
        alt="Detail"
        className="preview-photo"
        onClick={[Function]}
        onDoubleClick={[Function]}
        src={undefined}
        style={
          Object {
            "background": "#ffffff",
          }
        } />
    </div>
    <div
      className="view-button-zone">
      <button
        className="view-button"
        onClick={[Function]}
        style={
          Object {
            "opacity": "0",
          }
        }>
        View
      </button>
    </div>
  </div>
  <div
    className="selecting-photo-cell"
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}>
    <div
      className="preview-photo-zone">
      <img
        alt="Detail"
        className="preview-photo"
        onClick={[Function]}
        onDoubleClick={[Function]}
        src={undefined}
        style={
          Object {
            "background": "#ffffff",
          }
        } />
    </div>
    <div
      className="view-button-zone">
      <button
        className="view-button"
        onClick={[Function]}
        style={
          Object {
            "opacity": "0",
          }
        }>
        View
      </button>
    </div>
  </div>
</div>
`;
